base/io.jl

```diff
@@ -1003,6 +1000,46 @@ eltype(::Type{<:EachLine}) = String
 
 IteratorSize(::Type{<:EachLine}) = SizeUnknown()
 
+struct ReadEachIterator{T, IOT <: IO}
+    stream::IOT
+end
+
+"""
+    readeach(io::IO, T)
+
+Return an iterable object yielding [`read(io, T)`](@ref).
+
+See also: [`skipchars`](@ref), [`eachline`](@ref), [`readuntil`](@ref)
+
+!!! compat "Julia 1.6"
+    `readeach` requires Julia 1.6 or later.
+
+# Examples
+```jldoctest
+julia> open("my_file.txt", "w") do io
+           write(io, "JuliaLang is a GitHub organization.\\n It has many members.\\n");
```

Comment:
For demonstration purposes, maybe using `IOBuffer("JuliaLang is a GitHub organization.\\n It has many members.\\n")` would be enough as the focus is on `readeach` ?

Reply:
I copied the format of these doctests from the surrounding functions e.g. `readlines`. I'd prefer to keep them consistent, although I agree that using `IOBuffer` rather than opening an actual file might add to their clarity?